src/cpu_shader/path_tiling.rs

```diff
@@ -0,0 +1,152 @@
+// Copyright 2023 The Vello authors
+// SPDX-License-Identifier: Apache-2.0 OR MIT OR Unlicense
+
+use vello_encoding::{BumpAllocators, LineSoup, Path, PathSegment, SegmentCount, Tile};
+
+use crate::cpu_dispatch::CpuBinding;
+
+use super::util::{span, Vec2};
+
+const TILE_WIDTH: u32 = 16;
+const TILE_HEIGHT: u32 = 16;
+const TILE_SCALE: f32 = 1.0 / 16.0;
+
+fn path_tiling_main(
+    bump: &mut BumpAllocators,
+    seg_counts: &[SegmentCount],
+    lines: &[LineSoup],
+    paths: &[Path],
+    tiles: &[Tile],
+    segments: &mut [PathSegment],
+) {
+    for seg_ix in 0..bump.seg_counts {
+        let seg_count = seg_counts[seg_ix as usize];
+        let line = lines[seg_count.line_ix as usize];
+        let counts = seg_count.counts;
+        let seg_within_slice = counts >> 16;
+        let seg_within_line = counts & 0xffff;
+
+        // coarse rasterization logic
+        let p0 = Vec2::from_array(line.p0);
+        let p1 = Vec2::from_array(line.p1);
+        let is_down = p1.y >= p0.y;
+        let (mut xy0, mut xy1) = if is_down { (p0, p1) } else { (p1, p0) };
+        let s0 = xy0 * TILE_SCALE;
+        let s1 = xy1 * TILE_SCALE;
+        let count = span(s0.x, s1.x) + span(s0.y, s1.y) - 1;
+
+        let dx = (s1.x - s0.x).abs();
+        let dy = s1.y - s0.y;
+        let dy_dxdy = dy / (dx + dy);
+        let a = 1.0 - dy_dxdy;
```

Comment:
I have the same question as before (why the different formulations between WGSL and Rust?)